crates/db_schema/src/traits.rs

```diff
@@ -266,6 +266,25 @@ pub trait Blockable {
     Self: Sized;
 }
 
+pub trait Notable {
+  type Form;
+  type ObjectIdType;
+  type ObjectType;
+  fn note(
+    pool: &mut DbPool<'_>,
+    form: &Self::Form,
+  ) -> impl Future<Output = LemmyResult<Self>> + Send
+  where
+    Self: Sized;
+  fn delete_note(
+    pool: &mut DbPool<'_>,
+    person_id: PersonId,
+    target_id: Self::ObjectIdType,
+  ) -> impl Future<Output = LemmyResult<uplete::Count>> + Send
+  where
+    Self: Sized;
+}
```

Comment:
Theres no use in having a trait if its only implemented for a single struct.

Reply:
An extra trait also affects compilation time, so its better to leave it out until we actually use it.